real_intent/events/scrapy_claude/claude_sync.py

```
@@ -197,4 +203,5 @@ def perform_search(self, query: str, instance: Instance) -> ToolResult:
                 system=result.get("system") if result else None,
             )
         except Exception as e:
+            print("Error in search tool", e)
```

Remove `print`, or log instead. Also, why are we catching exception and then re-raising? Isn't context lost in the traceback?

---

yeah, doesn't make sense. I think we should remove try catch for tools, and in ToolCollection.run(), we let it raise the exception. wdyt

---

Yes like that a lot more 💯